python/mage/node_classification/utils/extract_from_database.py

```diff
@@ -0,0 +1,258 @@
+        if node_type == observed_attribute:
+            data[node_type].y[append_counter[node_type]] = int(
+                node.properties.get(class_name)
+            )
+
+        # increase append_counter by 1
+        append_counter[node_type] += 1
+
+    return data, reindexing, inv_reindexing, observed_attribute
+
+
+def edges_fetching(
+    ctx: mgp.ProcCtx, features_name: str, inv_reindexing: defaultdict, data: HeteroData
+) -> HeteroData:
+    """This procedure fetches the edges from the database and returns them in HeteroData object.
+
+    Args:
+        ctx: The context of the procedure.
+        features_name: The name of the database features attribute.
+        inv_reindexing: The inverse reindexing dictionary.
+        data: HeteroData object where edges will be stored.
+
+    Returns:
+        HeteroData object with edges.
+    """
+
+    edges = []  # variable for storing edges
+    edge_types = []  # variable for storing edge types
+    append_counter = defaultdict(int)  # variable for storing append counter
+
+    # obtain edges from context
+    for vertex in ctx.graph.vertices:
+        for edge in vertex.out_edges:
+            # edge_type is (from_vertex type, edge name, to_vertex type)
+            edge_type = tuple(
+                (
+                    edge.from_vertex.labels[0].name,
+                    edge.type.name,
+                    edge.to_vertex.labels[0].name,
+                )
+            )
+
+            if (
+                features_name not in edge.from_vertex.properties
+                or features_name not in edge.to_vertex.properties
+            ):
+                continue  # if from_vertex or out_vertex do not have features, skip the edge
+
+            edge_types.append(edge_type)  # append edge type to list of edge types
+            edges.append(edge)  # append edge to list of edges
+
+    edge_types = Counter(
+        edge_types
+    )  # apply Counter to obtain the number of each edge type
+
+    # set edge_index variables to empty tensors of size 2 x no_edge_type_edges
+    for edge_type, no_edge_type_edges in edge_types.items():
+        data[edge_type].edge_index = torch.tensor(
+            np.zeros((2, no_edge_type_edges)), dtype=torch.long
+        )
+
+    for edge in edges:
+        (from_vertex_type, edge_name, to_vertex_type) = (
+            edge.from_vertex.labels[0].name,
+            edge.type.name,
+            edge.to_vertex.labels[0].name,
+        )
+        edge_type = tuple((from_vertex_type, edge_name, to_vertex_type))
+
+        # add edge coordinates to edge_index tensors
+        data[edge_type].edge_index[0][append_counter[edge_type]] = int(
+            inv_reindexing[from_vertex_type][edge.from_vertex.id]
+        )
+        data[edge_type].edge_index[1][append_counter[edge_type]] = int(
+            inv_reindexing[to_vertex_type][edge.to_vertex.id]
+        )
+
+        append_counter[edge_type] += 1
+
+    return data
+
+
+def masks_generating(
```

Comment:
Rename function to generating_masks_for_X so it is clear what is purpose of function.